taca/illumina/HiSeq_Runs.py

```diff
@@ -661,13 +661,14 @@ def _generate_clean_samplesheet(self, ssparser):
                     entry[_data_filed_conversion(field)] = value
             data.append(entry)
 
-        fields_to_output = ['Lane', 'Sample_ID', 'Sample_Name', 'index', 'index2', 'Sample_Project']
+        fields_to_output = ['Lane', ssparser.dfield_sid, ssparser.dfield_snm, 'index', 'index2', ssparser.dfield_proj]
         #now create the new SampleSheet data section
         output+="[Data]{}".format(os.linesep)
         for field in ssparser.datafields:
-            new_field = _data_filed_conversion(field)
-            if new_field not in fields_to_output:
-                fields_to_output.append(new_field)
+            if field not in fields_to_output:
```

Comment:
Yes, for two reasons

1) since we thought of retaining the original header from the samplesheet we could avoid calling the `_data_filed_conversion` if the header is already in `fields_to_output`

2) the dictionary keys for `datafieldsConversion` ([here](https://github.com/SciLifeLab/TACA/blob/master/taca/illumina/HiSeq_Runs.py#L702)) in `_data_filed_conversion()` is hard coded. In case the header for `HiSeq2500` is changed then it would `raise` a `RunTime` error while calling `_data_filed_conversion`